src/devtools/client/debugger/src/utils/editor/get-expression.js

```diff
@@ -32,7 +32,7 @@ export function getExpressionFromCoords(cm, coord) {
   while (startHighlight > 1 && line.charAt(startHighlight - 1) === ".") {
     const tokenBefore = tokenAtTextPosition(cm, {
       line: coord.line,
-      column: startHighlight - 2,
```

Comment:
At first glance this seemed right to me, but I guess it's an off-by-one error. It only shows up if the first variable in a property chain is one character long.

```js
// this works fine
one.two.three

// but this wouldn't highlight
o.two.three
```

☝️  the off-by-one would skip the `o` and hit the `" "` and cause us to bail out further down the loop.

https://github.com/RecordReplay/devtools/blob/dbdf738bdb3509165820b4dc5bd39145091c875b/src/devtools/client/debugger/src/utils/editor/get-expression.js#L38-L40

Reply:
Is it possible for this to end up with a negative index? Or, asked another way, can `startHighlight` ever be `1`?

Reply:
> Is it possible for this to end up with a negative index? Or, asked another way, can `startHighlight` ever be `1`?

The clause of the `while` loop ensures `startHighlight > 1` I think.

Reply:
☝️  yep what Brian said

Reply:
@colbyr I wonder if we can re-enable these unit tests https://github.com/RecordReplay/devtools/blob/master/src/devtools/client/debugger/src/utils/editor/tests/get-expression.spec.js

cc @jcmorrow who might know how. Perhaps change it to `.test.js`

Reply:
> The clause of the while loop ensures startHighlight > 1 I think.

🤦  yep, good call!

Reply:
And yeah, it might be possible to enable those tests, but you'll have to manually lift them out of `src/devtools`, as `jest` is currently set to ignore that whole folder (and they should be renamed to match the naming scheme as Jason mentioned).

Reply:
perhaps we can allow `src/devtools` but it has to use the new naming scheme...

Reply:
sure I can do that! I didn't even notice there was a test in there. thanks @jasonLaster

Reply:
https://github.com/RecordReplay/devtools/pull/4290 this will let jest run `.test.js` files in src/devtools.

Getting the [get-expression tests](https://github.com/RecordReplay/devtools/blob/master/src/devtools/client/debugger/src/utils/editor/tests/get-expression.spec.js) might be a different story though... CodeMirror doesn't seem to work very well in the jsdom environment 🙈

Reply:
@colbyr weird - this is what we had before... jest + jsdom, but yeah - feel free to punt if this is a rabbit hole

Reply:
Yeah I think this is beyond me right now. I spent a few minutes trying to get them running, but for whatever reason the outputs seem to be different than they are in the browser environment and they're failing with or without my code changes in this PR.

<img width="664" alt="One of the failing get-expression tests" src="https://user-images.githubusercontent.com/478109/140171112-510fe0e2-2cd7-44a8-ba33-96f10bdab4ac.png">

Idk how long its been since these tests were running, but maybe something in jsdom/codemirror/jest has changed since then. If anyone is hankering to get into it, I put up a branch where I was at: https://github.com/RecordReplay/devtools/compare/master...colbyr:bring-back-get-expression-tests

Reply:
cool - lets skip it.